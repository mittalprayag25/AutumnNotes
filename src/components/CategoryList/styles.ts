import {StyleSheet} from 'react-native';
import {sizes} from './../../constants/sizes';
import {colors} from './../../constants/colors';

const styles = StyleSheet.create({
  container: {
    //height: '20%',
  },
  header: {
    fontSize: sizes.font.md,
  },
  listItem: {
    height: 128,
    width: '45%',
    margin: sizes.space.sm,
    backgroundColor: colors.grey,
    alignItems: 'flex-start',
    padding: sizes.space.sm,
    borderRadius: sizes.borderRadius.sm,
  },
  categoryLabel: {
    fontSize: sizes.font.lg,
    marginLeft: sizes.space.sm,
    fontWeight: 'bold',
  },
  categoryGroup: {
    backgroundColor: colors.group,
    marginVertical: sizes.space.sm,
  },
});

export default styles;
