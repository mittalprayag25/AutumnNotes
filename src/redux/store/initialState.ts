const initialState = {
  notes: {
    allNotes: [],
    singleNote: null,
  },
  categories: {
    categories: [],
  },
};

export default initialState;
