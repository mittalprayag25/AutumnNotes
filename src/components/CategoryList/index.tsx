import React from 'react';
import {FlatList, Text, TouchableOpacity, View} from 'react-native';
import {Routes} from '../../constants/NavigationUtils';
import styles from './styles';

const CategoryList = ({navigation, data}: any) => {
  const renderItem = ({item, index}) => {
    return (
      <TouchableOpacity
        style={styles.listItem}
        testID={`note-detail-item-${item.title}`}
        onPress={() =>
          navigation.navigate(Routes.NOTEDETAIL, {
            singleNote: item,
            noteList: [],
            index: index,
          })
        }>
        <Text style={styles.header}>{`${item.title}`}</Text>
      </TouchableOpacity>
    );
  };

  return (
    <View style={styles.categoryGroup}>
      <Text style={styles.categoryLabel}>
        {data.category.length > 0 ? data.category : 'Misc'}
      </Text>
      <FlatList
        style={styles.container}
        data={data.data}
        renderItem={renderItem}
        numColumns={2}
      />
    </View>
  );
};

export default CategoryList;
