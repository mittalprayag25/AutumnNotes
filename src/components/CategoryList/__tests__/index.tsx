import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import * as reactRedux from 'react-redux';
import {Provider} from 'react-redux';

import {
  createMockStore,
  mockNavigationWithParams,
} from '../../../__mocks__/index';
import CategoryList from '../index';
import initialState from './../../../redux/store/initialState';

const mockedDispatch = jest.fn();

describe('Category List with some category', () => {
  const navigation = mockNavigationWithParams({});
  const props = {
    navigation: navigation,
    data: {
      category: 'category',
      data: [
        {category: 'category', title: 'title', description: 'description'},
      ],
    },
  };

  const changeState = {
    ...initialState,
  };
  it('should match snapshot', () => {
    const wrapper = render(<CategoryList {...props} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe('Category List with no category', () => {
  const navigation = mockNavigationWithParams({});
  const props = {
    navigation: navigation,
    data: {
      category: '',
      data: [
        {category: 'category', title: 'title', description: 'description'},
      ],
    },
  };

  const changeState = {
    ...initialState,
  };
  it('should match snapshot', () => {
    const wrapper = render(<CategoryList {...props} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should go back', () => {
    const {getByTestId} = render(<CategoryList {...props} />);
    const item = getByTestId('note-detail-item-title');
    act(() => {
      fireEvent(item, 'press');
    });
    expect(navigation.navigate).toHaveBeenCalledTimes(1);
  });
});
