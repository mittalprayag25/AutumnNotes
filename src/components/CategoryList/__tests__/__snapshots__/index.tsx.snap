// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category List with no category should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeaaa",
      "marginVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginLeft": 8,
      }
    }
  >
    Misc
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "category": "category",
          "description": "description",
          "title": "title",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={Object {}}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "#eeeee4",
                "borderRadius": 8,
                "height": 128,
                "margin": 8,
                "opacity": 1,
                "padding": 8,
                "width": "45%",
              }
            }
            testID="note-detail-item-title"
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              title
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Category List with some category should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeaaa",
      "marginVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginLeft": 8,
      }
    }
  >
    category
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "category": "category",
          "description": "description",
          "title": "title",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={Object {}}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "#eeeee4",
                "borderRadius": 8,
                "height": 128,
                "margin": 8,
                "opacity": 1,
                "padding": 8,
                "width": "45%",
              }
            }
            testID="note-detail-item-title"
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              title
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
